frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -311,20 +318,41 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
 
     const ocsObj = _.cloneDeep(ocsRequestData);
     ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
-    promises.push(k8sCreate(OCSServiceModel, ocsObj));
-
-    Promise.all(promises)
-      .then(() => {
-        history.push(
-          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
-            ocsProps.clusterServiceVersion.metadata.name
-          }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
-        );
-      })
-      .catch((err) => {
-        setProgress(false);
-        setError(err.message);
-      });
+
+    if (promises) {
+      Promise.all(promises)
+        .then(() => {
+          k8sCreate(OCSServiceModel, ocsObj)
+            .then(() => {
+              history.push(
+                `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+                  ocsProps.clusterServiceVersion.metadata.name
+                }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
```

Comment:
use `resourcePathFromModel`

Reply:
@bipuladh We need `ocs.openshift.io~v1~StorageCluster` here, hence `referenceForModel` is being used. Output of `resourcePathFromModel` is `/k8s/all-namespaces/ocs.openshift.io~v1~StorageCluster` which is not needed here

Reply:
resourcePathFromModel would replace the whole of the line and you can specify the namespace as per your need as well as the appName.

Reply:
@bipuladh resourcePathFromModel would not give us the the same path. I do feel like maybe the path used here might actually be a bug. The path to k8s resources should be standardized, and this appears at first glance to be different from the usual format.